src/patterns/api-lambda.test.ts

```diff
@@ -53,4 +56,46 @@ describe("The GuApiLambda pattern", () => {
     });
     expect(stack).toHaveResource("AWS::CloudWatch::Alarm");
   });
+
+  it("should allow us to link a domain name to a LambdaRestApi", () => {
+    const stack = simpleGuStackForTesting();
+    const noMonitoring: NoMonitoring = { noMonitoring: true };
+    const apiLambda = new GuApiLambda(stack, "lambda", {
+      fileName: "my-app.zip",
+      handler: "handler.ts",
+      runtime: Runtime.NODEJS_12_X,
+      monitoringConfiguration: noMonitoring,
+      app: "testing",
+      apis: [
+        {
+          id: "api",
+          description: "this is a test",
+        },
+        {
+          id: "api2",
+          description: "this is a test2",
+        },
+      ],
+    });
+    const api = apiLambda.apis.get("api");
+    expect(api).toBeDefined();
+    /* eslint-disable @typescript-eslint/no-non-null-assertion -- We expect api to be defined beyond this point */
+    api!.addDomainName("domain", {
+      domainName: "code.theguardian.com",
+      certificate: new GuCertificate(stack, {
+        app: "testing",
+        domainName: "code.theguardian.com",
+        hostedZoneId: "id123",
+      }),
+    });
+
+    new GuCname(stack, "DNS", {
+      app: "testing",
+      ttl: Duration.days(1),
+      domainName: "code.theguardian.com",
+      resourceRecord: api!.domainName!.domainNameAliasDomainName,
```

Comment:
Curious why `domainName` is nullable here, I guess that's just how AWS have typed it?

Reply:
I think that's correct - it looks like it's an optional value https://github.com/aws/aws-cdk/blob/v1.137.0/packages/@aws-cdk/aws-apigateway/lib/restapi.ts#L150